File: ga/gpt3_5_result_2024_6/result/macro_func_gpt/12_macro_func.cpp
use std::io;

fn cpp_max(x: i32, y: i32) -> i32 {
    if x + y < 5 {
        return y;
    } else {
        return x;
    }
}

fn main() {
    let mut a = String::new();
    let mut b = String::new();

    io::stdin().read_line(&mut a)
        .expect("Failed to read input.");
    
    io::stdin().read_line(&mut b)
        .expect("Failed to read input.");
    
    let a: i32 = a.trim().parse()
        .expect("Please enter a valid number.");
    let b: i32 = b.trim().parse()
        .expect("Please enter a valid number.");

    let result = cpp_max(a, b);

    println!("Result: {}", result);
}

File: ga/glm4_result_2024_6/result/macro_func_glm/12_macro_func.rs
use std::io;

fn cppmax(x: i32, y: i32) -> i32 {
    if (x + y) < 5 {
        y
    } else {
        x
    }
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let mut parts = input.trim().split_whitespace();
    let a: i32 = parts.next().unwrap().parse().unwrap();
    let b: i32 = parts.next().unwrap().parse().unwrap();

    let result = cppmax(a, b);
    println!("Result: {}", result);
}